src/main/java/uk/gov/hmcts/ccd/domain/service/processor/SearchInputProcessor.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.hmcts.ccd.domain.service.processor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.data.casedetails.search.MetaData;
+import uk.gov.hmcts.ccd.domain.model.definition.FieldType;
+import uk.gov.hmcts.ccd.domain.model.search.CriteriaInput;
+import uk.gov.hmcts.ccd.domain.model.search.CriteriaType;
+import uk.gov.hmcts.ccd.domain.service.aggregated.DefaultGetCriteriaOperation;
+import uk.gov.hmcts.ccd.domain.service.aggregated.GetCriteriaOperation;
+import uk.gov.hmcts.ccd.endpoint.exceptions.DataProcessingException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+public class SearchInputProcessor {
```

Comment:
this is not a generic processor, it's tied to doing date transformation so we should rename it to DateSearchInputProcessor or sim